app/graphql/mutations/create_grading.rb

```diff
@@ -38,48 +38,35 @@ def resolve(_params)
     end
 
     class ValidateReviewData < GraphQL::Schema::Validator
-      include ValidatorCombinable
       def validate(_object, _context, value)
-        @submission_id = value[:submission_id]
         @submission = TimelineEvent.find_by(id: value[:submission_id])
         @checklist = value[:checklist]
-        @evaluation_criteria = @submission.evaluation_criteria
+        @evaluation_criteria = @submission&.evaluation_criteria
         @grades = value[:grades]
-        combine(
-          submission_exists,
-          submission_not_reviewed,
-          valid_evaluation_criteria,
-          right_shape_for_checklist,
-          checklist_data_is_not_mutated,
-          valid_grading
-        )
-      end
 
-      def submission_exists
-        return if @submission.present?
+        grade_hash = compute_grade_hash
 
-        I18n.t(
-          "mutations.create_grading.submission_missing_error",
-          submission_id: @submission_id
-        )
+        assignment_must_be_graded(value[:submission_id]) ||
+          checklist_must_have_right_shape || submission_must_not_be_reviewed ||
+          checklist_data_should_not_be_mutated ||
+          grading_should_be_valid(grade_hash)
       end
 
-      def submission_not_reviewed
+      def submission_must_not_be_reviewed
         return unless @submission.reviewed?
 
-        I18n.t('mutations.create_grading.submission_reviewed_error')
+        I18n.t("mutations.create_grading.submission_reviewed_error")
       end
 
-      def valid_evaluation_criteria
+      def assignment_must_be_graded(submission_id)
```

Comment:
@kaisersakhi Why is this function renamed to `assignment_must_be_graded`? That sounds confusing. It is actually just checking whether there is an evaluation criteria right?

Reply:
@kaisersakhi I agree here. It should be something on the lines of `assignment_is_reviewable` which pertains to having an evaluation criteria